datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -390,6 +433,63 @@ impl AsyncFileReader for ParquetFileReader {
     }
 }
 
+impl Display for DefaultParquetFileReaderFactory {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "DefaultParquetFileReaderFactory")
+    }
+}
+
+impl Debug for DefaultParquetFileReaderFactory {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "DefaultParquetFileReaderFactory")
+    }
+}
+
+impl ParquetFileReaderFactory for DefaultParquetFileReaderFactory {
+    fn create_reader(
+        &self,
+        file_meta: FileMeta,
+        metadata_size_hint: Option<usize>,
+        metrics: ParquetFileMetrics,
+    ) -> Result<Box<dyn AsyncFileReader + Send>> {
+        Ok(Box::new(ParquetFileReader {
+            meta: file_meta.object_meta,
+            store: Arc::clone(&self.store),
+            metadata_size_hint,
+            metrics,
+        }))
+    }
+}
+
+struct ThinFileReader {
```

Comment:
We should probably fix the need for this upstream. In the meantime perhaps we could called this `BoxedAsyncFileReader` to make clear what it is for

Reply:
sure, by the way do you know any better way to solve that than for example making upstream accept `Box<dyn AsyncFileReader + Send>` in constructor ?

Reply:
Within `parquet` we could implement `AsyncFileReader for Box<dyn AsyncFileReader>`

Reply:
Fix in https://github.com/apache/arrow-rs/pull/2368